src/main/java/logic/Storage.java

```diff
@@ -0,0 +1,107 @@
+package logic;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import exception.InvalidSaveFileException;
+import tasks.Deadlines;
+import tasks.Events;
+import tasks.Task;
+import tasks.ToDos;
+
+/**
+ * Handles saving of the list and reading it upon startup.
+ */
+public class Storage {
+    private String fileLocation;
+    public Storage(String fileLocation) {
+        this.fileLocation = fileLocation;
+    }
+
+    /**
+     * Called upon startup. Reads save file if there is one, else creates
+     * a blank save file.
+     *
+     * @return A List populated by the tasks.Task from previous save file.
+     * @throws InvalidSaveFileException If there is an issue with creating the save file.
+     */
+    public List<Task> readFile() throws InvalidSaveFileException {
+        List<Task> toDoList = new ArrayList<Task>();
+        File dir = new File("data");
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+        try {
+            File file = new File(fileLocation);
+            if (file.exists()) {
+                Scanner s = new Scanner(file);
+                while (s.hasNext()) {
+                    String entry = s.nextLine();
+                    DateTimeFormatter dtf2 = DateTimeFormatter.ofPattern("HH:mm MMM d yyyy");
+                    if (entry.startsWith("[T]")) {
```

Comment:
Would it be better to abstract `[T]` out into a constant variable within the `ToDo` class? Likewise for `[D]`